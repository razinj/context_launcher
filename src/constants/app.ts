// Modules
import AppsModule from '../native-modules/AppsModule'

// Constant values
export const {
  packageName: CONTEXT_LAUNCHER_APP_ID,
  appVersion: CONTEXT_LAUNCHER_APP_VERSION,
  buildNumber: CONTEXT_LAUNCHER_APP_BUILD_NUMBER,
} = AppsModule.getConstants()
export const APP_ITEM_HEIGHT_ICON_DISPLAYED = 60
// Colors
export const PRIMARY_HEX_COLOR = '#42855B'
export const SECONDARY_HEX_COLOR = '#A0C2AD'
export const BACKGROUND_COLOR = 'rgba(255, 255, 255, 0.2)'
// Native
export const HARDWARE_BACK_PRESS_EVENT_NAME = 'hardwareBackPress'
// React Native
export const KEYBOARD_DID_SHOW_EVENT_NAME = 'keyboardDidShow'
export const KEYBOARD_DID_HIDE_EVENT_NAME = 'keyboardDidHide'
