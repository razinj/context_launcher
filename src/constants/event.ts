export const PACKAGE_CHANGE_EVENT_NAME = 'packageChange'
export const PACKAGE_CHANGE_EVENT_NAME_X = 'packageChangeX'
