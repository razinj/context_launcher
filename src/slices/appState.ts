import { createAction, createSlice, PayloadAction } from '@reduxjs/toolkit'
import { AppDetails } from '../models/app-details'
import { RenderedIn } from '../models/rendered-in'
import { RootState } from '../store'

interface AppsSearchState {
  query?: string
  result: AppDetails[]
}

export interface AppState {
  displayAllApps: boolean
  displaySettings: boolean
  menuAppDetails?: AppDetails
  displayAppMenu: boolean
  displaySortableFavoriteApps: boolean
  displaySortablePinnedApps: boolean
  displaySortableTemporaryPinnedApps: boolean
  search: AppsSearchState
}

const initialState: AppState = {
  displayAllApps: false,
  displaySettings: false,
  menuAppDetails: undefined,
  displayAppMenu: false,
  displaySortableFavoriteApps: false,
  displaySortablePinnedApps: false,
  displaySortableTemporaryPinnedApps: false,
  search: {
    query: undefined,
    result: [],
  },
}

export const appStateSlice = createSlice({
  name: 'appState',
  initialState,
  reducers: {
    // GlobalAppState
    setDisplayAllApps: (state: AppState, { payload }: PayloadAction<boolean>) => {
      state.displayAllApps = payload
    },
    setDisplaySettings: (state: AppState, { payload }: PayloadAction<boolean>) => {
      state.displaySettings = payload
    },
    setMenuAppDetails: (state: AppState, { payload }: PayloadAction<AppDetails | undefined>) => {
      state.menuAppDetails = payload
    },
    setDisplayAppMenu: (state: AppState, { payload }: PayloadAction<boolean>) => {
      state.displayAppMenu = payload
    },
    setDisplaySortableFavoriteApps: (state: AppState, { payload }: PayloadAction<boolean>) => {
      state.displaySortableFavoriteApps = payload
    },
    setDisplaySortablePinnedApps: (state: AppState, { payload }: PayloadAction<boolean>) => {
      state.displaySortablePinnedApps = payload
    },
    setDisplaySortableTemporaryPinnedApps: (state: AppState, { payload }: PayloadAction<boolean>) => {
      state.displaySortableTemporaryPinnedApps = payload
    },
    // AppsSearchState
    setAppsSearchQuery: (state: AppState, { payload }: PayloadAction<string | undefined>) => {
      state.search.query = payload
    },
    setAppsSearchResult: (state: AppState, { payload }: PayloadAction<AppDetails[]>) => {
      state.search.result = payload
    },
    resetAppsSearchState: (state: AppState) => {
      state.search.query = undefined
      state.search.result = []
    },
  },
})

export const {
  setDisplayAllApps,
  setDisplaySettings,
  setMenuAppDetails,
  setDisplayAppMenu,
  setDisplaySortableFavoriteApps,
  setDisplaySortablePinnedApps,
  setDisplaySortableTemporaryPinnedApps,
  setAppsSearchQuery,
  setAppsSearchResult,
  resetAppsSearchState,
} = appStateSlice.actions

export const toogleAllApps = createAction('appState/toogleAllApps')
export const appLaunch = createAction<{ renderedIn: RenderedIn; appDetails: AppDetails }>('appState/appLaunch')
export const appLaunchFromSearch = createAction('appState/appLaunchFromSearch')
export const sortFavoriteApps = createAction('appState/sortFavoriteApps')
export const sortPinnedApps = createAction('appState/sortPinnedApps')
export const sortTemporaryPinnedApps = createAction('appState/sortTemporaryPinnedApps')

export const selectAppState = (state: RootState) => state.appState
export const selectDisplayAllApps = (state: RootState) => state.appState.displayAllApps
export const selectDisplaySettings = (state: RootState) => state.appState.displaySettings
export const selectMenuAppDetails = (state: RootState) => state.appState.menuAppDetails
export const selectDisplayAppMenu = (state: RootState) => state.appState.displayAppMenu
export const selectDisplaySortableFavoriteApps = (state: RootState) => state.appState.displaySortableFavoriteApps
export const selectDisplaySortablePinnedApps = (state: RootState) => state.appState.displaySortablePinnedApps
export const selectDisplaySortableTemporaryPinnedApps = (state: RootState) => state.appState.displaySortableTemporaryPinnedApps
export const selectAppsSearchQuery = (state: RootState) => state.appState.search.query
export const selectAppsSearchResult = (state: RootState) => state.appState.search.result

export default appStateSlice.reducer
