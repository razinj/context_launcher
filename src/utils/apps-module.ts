// Native modules
import AppsModule from '../native-modules/AppsModule'

export const launchApp = (packageName: string) => AppsModule.launchApplication(packageName)

export const showAppDetails = (packageName: string) => AppsModule.showApplicationDetails(packageName)

export const requestAppUninstall = (packageName: string) => AppsModule.requestApplicationUninstall(packageName)
