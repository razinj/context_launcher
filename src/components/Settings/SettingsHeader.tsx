import React from 'react'
import { Pressable, PressableAndroidRippleConfig, StyleSheet, Text, View } from 'react-native'
import { useDispatch } from 'react-redux'
import { APP_BUILD_NUMBER, APP_ID, APP_VERSION } from '../../constants'
import { setDisplaySettings } from '../../slices/appState'
import { showAppDetails } from '../../utils/apps-module'
import CustomIcon from '../shared/CustomIcon'

const appInfoIconRippleConfig: PressableAndroidRippleConfig = {
  borderless: true,
  foreground: true,
  color: '#e5e5e5',
  radius: 18,
}

const SettingsHeader = () => {
  const dispatch = useDispatch()

  const onAppInfoClick = () => {
    showAppDetails(APP_ID)
    dispatch(setDisplaySettings(false))
  }

  return (
    <View style={styles.wrapper} testID='settings-header-wrapper'>
      <Text style={styles.title}>
        Context Settings
        <Text style={styles.appInfoText}>
          &nbsp;&nbsp;v{APP_VERSION} ({APP_BUILD_NUMBER})
        </Text>
      </Text>
      <Pressable
        onPress={onAppInfoClick}
        android_disableSound={true}
        android_ripple={appInfoIconRippleConfig}
        testID='settings-header-app-info-button'>
        <CustomIcon name='information-outline' size={34} color='#808080' />
      </Pressable>
    </View>
  )
}

const styles = StyleSheet.create({
  wrapper: {
    paddingBottom: 20,
    alignItems: 'center',
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  title: {
    fontSize: 20,
    color: '#808080',
    fontWeight: '400',
  },
  appInfoText: {
    color: '#ccc',
    fontSize: 12,
  },
})

export default SettingsHeader
