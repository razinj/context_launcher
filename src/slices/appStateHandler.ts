import { PayloadAction } from '@reduxjs/toolkit'
import { call, put, select, takeLatest } from 'redux-saga/effects'
import { AppDetails } from '../models/app-details'
import { FavoriteApp } from '../models/favorite-app'
import { PinnedApp } from '../models/pinned-app'
import { RenderedIn } from '../models/rendered-in'
import { launchApp } from '../utils/apps-module'
import { dismissKeyboard } from '../utils/keyboard'
import {
  appLaunch,
  appLaunchFromSearch,
  resetAppsSearchState,
  selectAppsSearchResult,
  selectDisplayAllApps,
  setDisplayAllApps,
  setDisplaySettings,
  setDisplaySortableFavoriteApps,
  setDisplaySortablePinnedApps,
  setDisplaySortableTemporaryPinnedApps,
  setMenuAppDetails,
  sortFavoriteApps,
  sortPinnedApps,
  sortTemporaryPinnedApps,
  toogleAllApps,
} from './appState'
import { selectFavoriteApps } from './favoriteApps'
import { selectPinnedApps } from './pinnedApps'
import { addRecentApp } from './recentApps'

// We should only add a launched app if it was clicked (rendered) in one of the following places;
const ADD_TO_RECENT_APPS_RENDERED_IN_VALUES = [RenderedIn.ALL_APPS, RenderedIn.FILTERED_APPS]

function* toggleAllAppsHandler() {
  const allAppsDisplayed: boolean = yield select(selectDisplayAllApps)
  yield put(setDisplayAllApps(!allAppsDisplayed))
}

function* appLaunchHandler(action: PayloadAction<{ renderedIn: RenderedIn; appDetails: AppDetails }>) {
  yield call(dismissKeyboard)
  yield put(resetAppsSearchState())
  yield put(setDisplayAllApps(false))
  yield put(setMenuAppDetails(undefined))
  yield put(setDisplaySettings(false))
  yield put(setDisplaySortableFavoriteApps(false))

  const { appDetails, renderedIn } = action.payload

  if (!ADD_TO_RECENT_APPS_RENDERED_IN_VALUES.includes(renderedIn)) {
    return
  }

  // Skip adding to recent apps if permanently pinned/favourited but launched from different lists/views
  const pinnedApps: PinnedApp[] = yield select(selectPinnedApps)
  const favoriteApps: FavoriteApp[] = yield select(selectFavoriteApps)
  // TODO: Include the temporarily pinned apps but only when they're rendered/displayed.

  const pinnedAppsPackages: string[] = pinnedApps.map(({ packageName }: PinnedApp) => packageName)
  const favoriteAppsPackages: string[] = favoriteApps.map(({ packageName }: FavoriteApp) => packageName)
  const allApps = new Set([...pinnedAppsPackages, ...favoriteAppsPackages])

  if (allApps.has(appDetails.packageName)) {
    return
  }

  // Otherwise add to recent apps
  yield put(addRecentApp(appDetails))
}

function* sortFavoriteAppsHandler() {
  yield put(setDisplaySettings(false))
  yield put(setDisplaySortableFavoriteApps(true))
}

function* sortPinnedAppsHandler() {
  yield put(setDisplaySettings(false))
  yield put(setDisplaySortablePinnedApps(true))
}

function* sortTemporaryPinnedAppsHandler() {
  yield put(setDisplaySettings(false))
  yield put(setDisplaySortableTemporaryPinnedApps(true))
}

function* appLaunchFromSearchHandler() {
  const searchResult: AppDetails[] = yield select(selectAppsSearchResult)
  if (searchResult.length === 0) {
    return
  }

  const appDetails = searchResult[0]

  yield call(launchApp, appDetails.packageName)
  yield put(appLaunch({ renderedIn: RenderedIn.FILTERED_APPS, appDetails }))
}

export function* appStateSaga() {
  yield takeLatest(toogleAllApps.type, toggleAllAppsHandler)
  yield takeLatest(appLaunch.type, appLaunchHandler)
  yield takeLatest(sortFavoriteApps.type, sortFavoriteAppsHandler)
  yield takeLatest(sortPinnedApps.type, sortPinnedAppsHandler)
  yield takeLatest(sortTemporaryPinnedApps.type, sortTemporaryPinnedAppsHandler)
  yield takeLatest(appLaunchFromSearch.type, appLaunchFromSearchHandler)
}
