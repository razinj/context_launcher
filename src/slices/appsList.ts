import { createAction, createSelector, createSlice, PayloadAction } from '@reduxjs/toolkit'
import { AppDetails } from '../models/app-details'
import { RootState } from '../store'
import { getAppsLetterIndex } from '../utils/alphabet-list'
import { displayToast } from '../utils/toast'

export interface AppsListState {
  list: AppDetails[]
}

const initialState: AppsListState = {
  list: [],
}

export const appsListSlice = createSlice({
  name: 'appsList',
  initialState,
  reducers: {
    setAppsList: (state: AppsListState, { payload }: PayloadAction<AppDetails[]>) => {
      // console.log('setAppsList called');
      displayToast(`apps found : ${payload.length}`)

      state.list = payload.sort((appOne: AppDetails, appTwo: AppDetails) => appOne.name.localeCompare(appTwo.name))
    },
  },
})

export const { setAppsList } = appsListSlice.actions

export const getAppsListAction = createAction('appsList/getAppsListAction')
export const appRemovedAction = createAction<string>('appsList/appRemovedAction')
export const revalidateAppsLists = createAction('appsList/revalidateAppsLists')

const selectAppsList = (state: RootState) => state.appsList.list

export const selectAppsListMemoized = createSelector(selectAppsList, (list: AppDetails[]) => list)
export const selectAppsLetterListMemoized = createSelector(selectAppsList, (list: AppDetails[]) =>
  getAppsLetterIndex(list)
)

export default appsListSlice.reducer
