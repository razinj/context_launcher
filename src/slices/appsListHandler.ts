import { PayloadAction } from '@reduxjs/toolkit'
import { Channel, channel } from 'redux-saga'
import { call, fork, put, select, take, takeLatest } from 'redux-saga/effects'
import { AppDetails } from '../models/app-details'
import LauncherAppsModule from '../native-modules/LauncherAppsModule'
import { appRemovedAction, getAppsListAction, revalidateAppsLists, setAppsList } from './appsList'
import { removeFavoriteApp, removeFavoriteApps, selectFavoriteAppsMemoized } from './favoriteApps'
import {
  removePinnedApp,
  removePinnedApps,
  selectPinnedAppsMemoized,
  selectTemporaryPinnedAppsMemoized,
} from './pinnedApps'
import { removeRecentApp, removeRecentApps, selectRecentAppsMemoized } from './recentApps'

const notAvailable = 'NOT_AVAILABLE'

function* getAppsListActionHandler() {
  try {
    const applications: string = yield call(LauncherAppsModule.getApplications)
    const apps = JSON.parse(applications) as AppDetails[]
    yield put(setAppsList(apps))
  } catch (error: unknown) {
    yield put(setAppsList([]))
    console.error('Error getting apps list:', error)
  }
}

function* appRemovedActionHandler(action: PayloadAction<string>) {
  yield put(removeRecentApp(action.payload))
  const pinnedApp = { id: notAvailable, packageName: action.payload, name: notAvailable, icon: notAvailable }
  yield put(removePinnedApp({ app: pinnedApp, isPermanent: true })) // TODO: Check why I send an "app" inseatd of "packageName" for removal
  yield put(removePinnedApp({ app: pinnedApp, isPermanent: false }))
  yield put(removeFavoriteApp(action.payload))
}

// Revalidate all apps' lists
function* x() {
  console.log('x called')

  const applications: string = yield call(LauncherAppsModule.getApplications)
  const apps = JSON.parse(applications) as AppDetails[]
  yield put(setAppsList(apps))

  const appsPackageNames = apps.map(({ packageName }: AppDetails) => packageName)

  // Revalidate recent apps
  const recentApps: AppDetails[] = yield select(selectRecentAppsMemoized)
  yield put(
    removeRecentApps(
      recentApps
        .map(({ packageName }: AppDetails) => packageName)
        .filter(
          (packageName: string) => !appsPackageNames.some((appsPackageName: string) => appsPackageName === packageName)
        )
    )
  )

  // Revalidate favorite apps
  const favoriteApps: AppDetails[] = yield select(selectFavoriteAppsMemoized)
  yield put(
    removeFavoriteApps(
      favoriteApps
        .map(({ packageName }: AppDetails) => packageName)
        .filter(
          (packageName: string) => !appsPackageNames.some((appsPackageName: string) => appsPackageName === packageName)
        )
    )
  )

  // Revalidate pinned apps
  const pinnedApps: AppDetails[] = yield select(selectPinnedAppsMemoized)
  yield put(
    removePinnedApps({
      isPermanent: true,
      packageNames: pinnedApps
        .map(({ packageName }: AppDetails) => packageName)
        .filter(
          (packageName: string) => !appsPackageNames.some((appsPackageName: string) => appsPackageName === packageName)
        ),
    })
  )

  // Revalidate temporary pinned apps
  const temporaryPinnedApps: AppDetails[] = yield select(selectTemporaryPinnedAppsMemoized)
  yield put(
    removePinnedApps({
      isPermanent: false,
      packageNames: temporaryPinnedApps
        .map(({ packageName }: AppDetails) => packageName)
        .filter(
          (packageName: string) => !appsPackageNames.some((appsPackageName: string) => appsPackageName === packageName)
        ),
    })
  )
}

// const upsertAppsListChannel = channel()

// function* watchAppsListUpsert() {
//   while (true) {
//     type x = typeof setAppsList
//     const action: x = yield take(upsertAppsListChannel)
//     yield put(action)
//   }
// }

function getApplicationsV2Async() {
  return new Promise((resolve, reject) => {
    LauncherAppsModule.getApplicationsV2((applications: string) => {
      resolve(applications)
    })
  })
}

function* y() {
  console.log('y called')

  const applications: string = yield call(getApplicationsV2Async)
  yield call(handleApplications, applications)

  const apps = JSON.parse(applications) as AppDetails[]

  const appsPackageNames = apps.map(({ packageName }: AppDetails) => packageName)
  // const appsPackageNames: string[] = []

  // Revalidate recent apps
  const recentApps: AppDetails[] = yield select(selectRecentAppsMemoized)
  yield put(
    removeRecentApps(
      recentApps
        .map(({ packageName }: AppDetails) => packageName)
        .filter(
          (packageName: string) => !appsPackageNames.some((appsPackageName: string) => appsPackageName === packageName)
        )
    )
  )

  // Revalidate favorite apps
  const favoriteApps: AppDetails[] = yield select(selectFavoriteAppsMemoized)
  yield put(
    removeFavoriteApps(
      favoriteApps
        .map(({ packageName }: AppDetails) => packageName)
        .filter(
          (packageName: string) => !appsPackageNames.some((appsPackageName: string) => appsPackageName === packageName)
        )
    )
  )

  // Revalidate pinned apps
  const pinnedApps: AppDetails[] = yield select(selectPinnedAppsMemoized)
  yield put(
    removePinnedApps({
      isPermanent: true,
      packageNames: pinnedApps
        .map(({ packageName }: AppDetails) => packageName)
        .filter(
          (packageName: string) => !appsPackageNames.some((appsPackageName: string) => appsPackageName === packageName)
        ),
    })
  )

  // Revalidate temporary pinned apps
  const temporaryPinnedApps: AppDetails[] = yield select(selectTemporaryPinnedAppsMemoized)
  yield put(
    removePinnedApps({
      isPermanent: false,
      packageNames: temporaryPinnedApps
        .map(({ packageName }: AppDetails) => packageName)
        .filter(
          (packageName: string) => !appsPackageNames.some((appsPackageName: string) => appsPackageName === packageName)
        ),
    })
  )
}

function* handleApplications(applications: string) {
  console.log('handleApplications called')

  const apps = JSON.parse(applications) as AppDetails[]
  // dispatch an action here
  yield put(setAppsList(apps))
}

function* childSaga(channel: Channel<any>) {
  const applications: string = yield call(LauncherAppsModule.getApplications)
  const apps = JSON.parse(applications) as AppDetails[]
  // yield put(setAppsList(apps))
  // const result = yield call(someAsyncFunction);
  yield put(channel, apps) // Emit the result through the channel
}

function* z() {
  console.log('z called')

  const communicationChannel: Channel<AppDetails[]> = yield call(channel)
  yield fork(childSaga, communicationChannel)

  while (true) {
    const result: AppDetails[] = yield take(communicationChannel)
    console.log(' result: ', result.length)

    const appsPackageNames = result.map(({ packageName }: AppDetails) => packageName)
    // const appsPackageNames: string[] = []

    // Revalidate recent apps
    const recentApps: AppDetails[] = yield select(selectRecentAppsMemoized)
    yield put(
      removeRecentApps(
        recentApps
          .map(({ packageName }: AppDetails) => packageName)
          .filter(
            (packageName: string) =>
              !appsPackageNames.some((appsPackageName: string) => appsPackageName === packageName)
          )
      )
    )

    // Revalidate favorite apps
    const favoriteApps: AppDetails[] = yield select(selectFavoriteAppsMemoized)
    yield put(
      removeFavoriteApps(
        favoriteApps
          .map(({ packageName }: AppDetails) => packageName)
          .filter(
            (packageName: string) =>
              !appsPackageNames.some((appsPackageName: string) => appsPackageName === packageName)
          )
      )
    )

    // Revalidate pinned apps
    const pinnedApps: AppDetails[] = yield select(selectPinnedAppsMemoized)
    yield put(
      removePinnedApps({
        isPermanent: true,
        packageNames: pinnedApps
          .map(({ packageName }: AppDetails) => packageName)
          .filter(
            (packageName: string) =>
              !appsPackageNames.some((appsPackageName: string) => appsPackageName === packageName)
          ),
      })
    )

    // Revalidate temporary pinned apps
    const temporaryPinnedApps: AppDetails[] = yield select(selectTemporaryPinnedAppsMemoized)
    yield put(
      removePinnedApps({
        isPermanent: false,
        packageNames: temporaryPinnedApps
          .map(({ packageName }: AppDetails) => packageName)
          .filter(
            (packageName: string) =>
              !appsPackageNames.some((appsPackageName: string) => appsPackageName === packageName)
          ),
      })
    )

    // Use the result returned by the child saga
    // Non-blocking logic here

    // ...
  }

  console.log('done')
}

export function* appsListSagas() {
  yield takeLatest(getAppsListAction.type, getAppsListActionHandler)
  yield takeLatest(appRemovedAction.type, appRemovedActionHandler)
  // yield takeLatest(revalidateAppsLists.type, x)
  // yield takeLatest(revalidateAppsLists.type, z)
  yield takeLatest(revalidateAppsLists.type, z)
}
