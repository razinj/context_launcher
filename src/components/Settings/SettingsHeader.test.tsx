import { fireEvent, screen } from '@testing-library/react-native'
import React from 'react'
import { renderWithProvider } from '../../../utils/test/utils'
import { APP_ID } from '../../constants'
import { setDisplaySettings } from '../../slices/appState'
import * as AppsModule from '../../utils/apps-module'
import SettingsHeader from './SettingsHeader'

const useDispatchMock = jest.fn()

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: () => useDispatchMock,
}))

describe('<SettingsHeader /> Tests', () => {
  beforeAll(() => {
    jest.spyOn(AppsModule, 'showAppDetails')
  })

  it('should render correctly and match snapshot', () => {
    renderWithProvider(<SettingsHeader />)

    expect(screen.toJSON()).toMatchSnapshot()
    expect(screen.getByTestId('wrapper')).toBeOnTheScreen()
    expect(screen.getByTestId('app-info-button')).toBeOnTheScreen()
  })

  it('should call function to display settings bottom sheet when pressed', () => {
    renderWithProvider(<SettingsHeader />)

    const appInfoButton = screen.getByTestId('app-info-button')

    expect(appInfoButton).toBeOnTheScreen()

    fireEvent.press(appInfoButton)

    expect(useDispatchMock).toHaveBeenCalledWith(setDisplaySettings(false))
    expect(AppsModule.showAppDetails).toHaveBeenCalledWith(APP_ID)
  })
})
