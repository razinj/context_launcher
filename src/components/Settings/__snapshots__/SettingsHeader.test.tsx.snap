// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SettingsHeader /> Tests should not render modal when display settings value is false (default) 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingBottom": 20,
        }
      }
      testID="wrapper"
    >
      <Text
        style={
          {
            "color": "#808080",
            "fontSize": 20,
            "fontWeight": "400",
          }
        }
      >
        Context Settings
        <Text
          style={
            {
              "color": "#ccc",
              "fontSize": 12,
            }
          }
        >
            v
          1.4.8
           (
          12
          )
        </Text>
      </Text>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        testID="settings-header-app-info-button"
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "#808080",
                "fontSize": 34,
              },
              undefined,
              {
                "fontFamily": "Material Design Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          󰋽
        </Text>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;

exports[`<SettingsHeader /> Tests should render correctly and match snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingBottom": 20,
        }
      }
      testID="wrapper"
    >
      <Text
        style={
          {
            "color": "#808080",
            "fontSize": 20,
            "fontWeight": "400",
          }
        }
      >
        Context Settings
        <Text
          style={
            {
              "color": "#ccc",
              "fontSize": 12,
            }
          }
        >
            v
          1.4.8
           (
          12
          )
        </Text>
      </Text>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        testID="app-info-button"
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "#808080",
                "fontSize": 34,
              },
              undefined,
              {
                "fontFamily": "Material Design Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          󰋽
        </Text>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
